File: server/src/main/java/com/example/schoolsite/workWithDatabase/repo/UserRepository.java
package com.example.schoolsite.workWithDatabase.repo;

import com.example.schoolsite.entity.User;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.Optional;

@Repository
public interface UserRepository extends JpaRepository<User, Long> {
    User findByLoginAndPasswordAndRole(String login, String password, String role);
    boolean existsUserByLogin(String login);
    Optional<User> findByLogin(String login);
    Boolean existsByLogin(String login);
}

File: server/src/main/java/com/example/schoolsite/workWithDatabase/repo/UserRepository.kt
package com.example.schoolsite.workWithDatabase.repo

import com.example.schoolsite.entity.User
import org.springframework.data.jpa.repository.JpaRepository
import org.springframework.stereotype.Repository
import java.util.Optional

@Repository
interface UserRepository : JpaRepository<User?, Long?> {
    fun existsUserByLogin(login: String?): Boolean
    fun findByLogin(login: String?): Optional<User?>?
    fun existsByLogin(login: String?): Boolean?
}